### PR TITLE
chore: remove csi-node-plugin from the mayastor repo

### PR DESCRIPTION
**Pull request by tiagolobocastro**
The node-plugin should live in the control-plane repo instead.

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Review by dsavitskiy — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
Looks like nvmeadm has been replaced with libnvme-rs so in that I'll probably revert the submodule add, since we won't need nvmeadm here.

File: chart/templates/csi-daemonset.yaml [hunk 1]
```diff
@@ -1,128 +1,8 @@
...
-          type: Directory
-      - name: plugin-dir
-        hostPath:
-          path: /var/lib/kubelet/plugins/mayastor.openebs.io/
-          type: DirectoryOrCreate
-      - name: kubelet-dir
-        hostPath:
-          path: /var/lib/kubelet
-          type: Directory
+  name: csi-daemonset-hack
```
Comment (akhilerm) [inline chart/templates/csi-daemonset.yaml:4]: Why are we doing this? Cant the daemonset be removed from here?

File: mayastor/Cargo.toml [hunk 2]
```diff
@@ -114,5 +114,5 @@ version = "0.8.2"
 [dev-dependencies]
 assert_matches = "1.5.0"
 composer = { path = "../composer" }
-nvmeadm = { path = "../nvmeadm" }
+nvmeadm = { path = "../dependencies/nvmeadm" }
```
Comment (akhilerm) [inline mayastor/Cargo.toml:117]: Do we still have dependency on nvmeadm after using `libnvme-rs` crate

**Review by akhilerm — COMMENTED** [review]
@tiagolobocastro a few questions on the changes introduced.

File: chart/templates/csi-daemonset.yaml (hunk 1 above)
Comment (tiagolobocastro) [reply chart/templates/csi-daemonset.yaml:4]: Temporary not to break e2e pipeline

File: mayastor/Cargo.toml (hunk 2 above)
Comment (tiagolobocastro) [reply mayastor/Cargo.toml:117]: As per my comment I need to rebase and update the PR

**Comment by tiagolobocastro** [comment]
bors try

File: mayastor/Cargo.toml (hunk 2 above)
Comment (tiagolobocastro) [reply mayastor/Cargo.toml]: @akhilerm I've updated the PR

**Review by paulyoong — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**